opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -20,73 +20,80 @@
 public interface Tracer {
 
   /**
-   * Create, start, and return a new Span with the given `operationName`.
-   * An optional parent Span can be specified used to incorporate the newly-returned Span into an existing trace.
+   * Return a new SpanBuilder for a Span with the given `operationName`.
    *
-   * <p>Example:
+   * <p>A contrived example:
    * <pre>{@code
     Tracer tracer = ...
 
-    Span feed = tracer.buildSpan("GetFeed")
-                      .start();
+    Span parentSpan = tracer.buildSpan("DoWork")
+                            .start();
 
     Span http = tracer.buildSpan("HandleHTTPRequest")
-                      .withParent(feed)
+                      .withChildOf(parentSpan.context())
                       .withTag("user_agent", req.UserAgent)
                       .withTag("lucky_number", 42)
                       .start();
     }</pre>
    */
   SpanBuilder buildSpan(String operationName);
 
-  /** Takes two arguments:
-   *    a Span instance, and
-   *    a “carrier” object in which to inject that Span for cross-process propagation.
+  /**
+   * Inject a SpanContext into a `carrier` of a given `format`, presumably for propagation across process boundaries.
+   *
+   * <p>Example:
+   * <pre>{@code
+   * Tracer tracer = ...
+   * Span clientSpan = ...
+   * TextMapWriter httpHeaderWriter = new AnHttpHeaderCarrier(httpRequest);
+   * tracer.inject(span.context(), BuiltinFormats.HTTP_HEADER, httpHeaderWriter);
+   * }</pre>
    *
-   * A “carrier” object is some sort of http or rpc envelope, for example HeaderGroup (from Apache HttpComponents).
+   * @param spanContext the SpanContext instance to inject into the carrier
+   * @param format the Format of the carrier. See BuiltinFormats.
+   * @param carrier the carrier for the SpanContext state; when inject() returns, the Tracer implementation will have represented the SpanContext within `carrier`
    *
-   * Attempting to inject to a carrier that has been registered/configured to this Tracer will result in a
-   * IllegalStateException.
+   * All implementations must support, at minimum, the BuiltinFormats.
    *
-   * All implementations support at minimum the required carriers BinaryWriter and TextMapWriter.
+   * @see BuiltinFormats
    */
-  <T> void inject(Span span, T carrier);
+  <W> void inject(SpanContext spanContext, Format<?, W> format, W carrier);
```

Comment:
@adriancole re TextMapReader and HttpHeaderReader: the pattern that's emerging is that some systems (e.g., gRPC) support unrestricted, sane, key:value string metadata maps. And then HTTP is a total mess from a key-mangling standpoint... yet it's obviously vitally important.

So, the gradual consensus has been that HTTP is "special" in that OpenTracing will recognize it with its own carrier (or pair of reader/writer carriers, whatever). But that we should also require Tracers to implement something less annoying like TextMap.

PS: none of this is at all interesting or even relevant if OT users aren't using Baggage.